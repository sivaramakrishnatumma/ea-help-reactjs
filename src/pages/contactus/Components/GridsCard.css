.grid-card-grid {
  padding-top: 20px;
}

.grid-card-grid span .selected {
  background: #000;
  color: #fff;
}

.grid-card-label {
  font-weight: 400 !important;
  font-size: 24px !important;
  line-height: 32px;
  margin-bottom: 16px;
  padding-right: 0;
  margin-top: 30px;
  margin-bottom: 30px;
}