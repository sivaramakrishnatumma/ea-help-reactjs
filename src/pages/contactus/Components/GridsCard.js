import React from "react";

import "./GridsCard.css";

import Card from "../../../shared/components/UIElements/Card/Card";

const GridsCard = props => {
  const { items, selectedItem, title, onSelect } = props;

  return (
    <div className="grid-card-container">
      <div className="grid-card-label">Select {title}*</div>
      <div className="grid-card-grid">
        {items.map((item, index) => {
          return (
            <span key={index} onClick={() => onSelect(item)}>
              <Card
                title={item.name}
                className={
                  selectedItem && item.id === selectedItem.id ? "selected" : ""
                }
              />
            </span>
          );
        })}
      </div>
    </div>
  );
};

export default GridsCard;
