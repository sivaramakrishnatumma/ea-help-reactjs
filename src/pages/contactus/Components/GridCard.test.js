import React from "react";
import { mount } from "enzyme";
import GridsCard from "./GridsCard";

describe("GridsCard component test case", () => {
  it("should test GridsCard component with default props", () => {
    const props = {
      items: [
        {
          id: "test",
          name: "test-name",
          imageUrl: "test-url"
        }
      ],
      onSelect: jest.fn(),
      title: "test-title",
      selectedItem: {
        id: "test",
        name: "test-name",
        imageUrl: "test-url"
      }
    };
    const wrapper = mount(<GridsCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
