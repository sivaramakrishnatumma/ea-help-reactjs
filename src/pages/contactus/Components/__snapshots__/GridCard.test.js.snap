// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridsCard component test case should test GridsCard component with default props 1`] = `
<GridsCard
  items={
    Array [
      Object {
        "id": "test",
        "imageUrl": "test-url",
        "name": "test-name",
      },
    ]
  }
  onSelect={[MockFunction]}
  selectedItem={
    Object {
      "id": "test",
      "imageUrl": "test-url",
      "name": "test-name",
    }
  }
  title="test-title"
>
  <div
    className="grid-card-container"
  >
    <div
      className="grid-card-label"
    >
      Select 
      test-title
      *
    </div>
    <div
      className="grid-card-grid"
    >
      <span
        key="0"
        onClick={[Function]}
      >
        <Card
          className="selected"
          title="test-name"
        >
          <div
            className="title-card selected"
          >
            <span>
              test-name
            </span>
          </div>
        </Card>
      </span>
    </div>
  </div>
</GridsCard>
`;
